Header Navigation Implementation

jonathanmdinh opened the pull request:
- Background Image: Background images in the menu overlay change based on hover interactions with navigation items.
- Menu Overlay Toggle: Smooth fade-in/out effect implemented for full-screen menu overlay.
- Menu Layout: Layout of links change based on screen size to support mobile utilization.
- Logo is implemented.

jonathanmdinh committed:
implemented logo and navigation menu

jonathanmdinh committed:
fixed p tags using wrong font

jonathanmdinh committed:
fixed left-alignment of contact info on large-screens

jonathanmdinh committed:
reformatted code for readability and scalability

jonathanmdinh committed:
reformatted code to show navigation when page has primary_navigation

jonathanmdinh committed:
added background change on menu item hover.

jonathanmdinh committed:
added comments and updated code for readability

jonathanmdinh committed:
added comments and updated code for readability

jonathanmdinh committed:
added improved hover effect to menu items

jonathanmdinh committed:
reduced transition duration for hamburger menu

jonathanmdinh committed:
Removed desktop/mobile separation for images. Added one const for background image.

jonathanmdinh committed:
Added modularity to navigation.js through helper functions

jonathanmdinh committed:
Removed arrows from menu overlay

d-la commented on wp-content/themes/chrisdinh-photography/resources/views/partials/page-header.blade.php:
```diff
@@ -1,3 +1,4 @@
-<div class="page-header">
+<!--temporary disable for testing-->
+<!--<div class="page-header">
```
Why is this commented out? If we don't want to show the partial we can remove the `@include` directive in the respective view template

jonathanmdinh replied:
Fixed in commit 7ac047240ddd1a582928084d9870c3de295a6442

d-la commented on wp-content/themes/chrisdinh-photography/resources/views/sections/header.blade.php:
```diff
@@ -1,11 +1,58 @@
-  @endif
+@php
+// Fetch menu items from the 'primary_navigation' menu if it exists. If not, return an empty array.
+$menuItems = has_nav_menu('primary_navigation') ? wp_get_nav_menu_items('primary_navigation') : [];
```
Lets try to keep the view template as clean as possible. We don't need to fetch data here since we have Controllers to do that. You should be able to fetch your data and return it globally in the `app/View/Composers/App.php` file. Try fetching the data there and using the set variable in this template

jonathanmdinh replied:
Updated in commit c46bd74f58be3b9077393fe2e455a29e49ffdb56

d-la commented on wp-content/themes/chrisdinh-photography/resources/views/sections/header.blade.php:
```diff
@@ -1,11 +1,58 @@
+                        <!-- Contect Information section -->
+                        <div class="w-full flex flex-col justify-start items-start p-4 pt-24 text-white whitespace-nowrap lg:w-1/2 lg:pr-48 z-20">
+                            <!-- Displaying contact information using ACF fields, resorts to default values if unavailable -->
+                            <p class="text-2xl md:text-4xl">{{ $social_media_handle ?? '@itschrisdinh' }}</p>
```
These should link out to the respective platform no? the instagram handle should be able to link out to instagram. For the phone number you can allow users to make a call via `tel:` in the `href` property i think (double check this). The email can auto open an email application using `mailto:` in the `href` property as well

jonathanmdinh replied:
adjusted in commit 7ac047240ddd1a582928084d9870c3de295a6442

d-la replied:
Where are these variables coming from `$social_media_handle`, etc? If they aren't ACF fields yet i would add them as ACF fields.

I would also wrap conditionals around each link to make sure the value isnt empty before creating the a tag, and use the supplied value in the `href` property as well. That way if the phone number field gets updated in the admin, both the number displayed and the `href` are the same and you dont need to touch the code

jonathanmdinh replied:
added acf fields for contact information to a newly created options page in commit 58494d75f161f14158697767c295041540ae40e8

d-la replied:
Going to resolve this thread but add more comments to your commit

d-la commented on wp-content/themes/chrisdinh-photography/resources/views/sections/header.blade.php:
```diff
@@ -1,11 +1,58 @@
+                                    @endphp
+                                    <!-- Menu item with dynamic images for mobile and desktop -->
+                                    <li class="nav-item mb-6" data-image="{{ $navigation_image }}">
+                                        <a href="{{ $menuItem->url }}" class="text-5xl hover:text-6xl lg:text-5xl">
```
Lets add support for linking out of the site. You should be able to use `$menuItem->target` to do so. Make sure the `target` property is not added if the value is not `"_blank"` since it'll throw an ADA error.

Did you try adding a transition property to make hover effect smoother? Currently it snaps to a larger font which looks weird to me but lmk your thoughts

jonathanmdinh replied:
Added support for linking out of the site in commit adb28007203b656883ae9c97775899c40f091802

jonathanmdinh replied:
Added smoother hover effect to menu items in commit b8ccbe52ade79578c123664dca2c3d2e6c9cd604

d-la replied:
@jonathanmdinh in your linking out of the website commit you did it wrong. You're adding a `target="_blank"` to the class property. Remove it from the class and add it as an inline conditional of the element so it becomes a property

jonathanmdinh replied:
fixed in commit 38e50ae5c42367846214bf62d942fc53e7bd5b1c

d-la replied:
@jonathanmdinh You didnt use the correct property in your conditional. Making the links open in a new tab in the admin doesnt work on the front end. going to close this thread out and add a comment to the place in the file

d-la commented on wp-content/themes/chrisdinh-photography/resources/views/sections/header.blade.php:
```diff
@@ -1,11 +1,58 @@
+                                @foreach ($menuItems as $menuItem)
+                                    @php
+                                        // Retrieve custom fields for each menu item, if available
+                                        $navigation_image = get_field('navigation__image', $menuItem);
```
Can you move this logic into the Controller where you fetch the menu items as well

jonathanmdinh replied:
Updated in commit ec7f8ae0412a694789af831ab3c28d09130dd6f8

d-la reviewed (changes requested):
Some small changes here. The other thing to note is maybe aligning the hamburger with the content on mobile

![styling](https://github.com/jonathanmdinh/itschrisdinh.com/assets/37962617/0334fea1-88cf-4f0e-a1e8-5ffc424a5c70)

We dont _need_ to do this but i think aligning the content looks nicer. This isnt required to approve the PR, just a thought

jonathanmdinh committed:
adjusted padding on menu items

jonathanmdinh committed:
Converted contact info links. Removed page-header from front-page view template

jonathanmdinh committed:
Fetching menuItems data in controller to return globally

jonathanmdinh committed:
Added support for linking out of the site

jonathanmdinh committed:
removed hover effect

jonathanmdinh committed:
Added smooth hover effect to navigation menu items

jonathanmdinh committed:
Fetch navigation images in Controller and returning globally

jonathanmdinh committed:
implemented options page for global contact info

jonathanmdinh committed:
fixed linking out of site implementation

d-la commented on wp-content/themes/chrisdinh-photography/app/View/Composers/FrontPage.php:
```diff
@@ -24,6 +24,12 @@ private function getHomepageSliderSettings() {
         return get_field('homepage__slider-settings');
     }
 
+    private function getContactInformation() {
```
@jonathanmdinh If this information appears in the header, which is a global layout, this data will not become available to other places in the site unless we call the same function again in other Composer files because you've added it to the `FrontPage.php` Composer, which only sends data specific to the Homepage. This should be in `app/View/Composers/App.php` so its global

jonathanmdinh replied:
updated in commit 918c2c53e2ceeeda24c229189cca9c688621f929

d-la replied:
Looks good @jonathanmdinh

d-la commented on wp-content/themes/chrisdinh-photography/resources/views/sections/header.blade.php:
```diff
@@ -1,11 +1,61 @@
+                                    <li class="nav-item mb-6" data-image="{{ $menuItem->navigation_image }}">
+                                        <a href="{{ $menuItem->url }}"
+                                        class="inline-block text-5xl lg:text-5xl transition-transform duration-300 ease-in-out hover:scale-110"
+                                        @if ($menuItem->link_target === '_blank') target="_blank" rel="noopener noreferrer" @endif>
```
Setting a menu item in the admin to open in a new tab and then testing on the front end does not actually open the menu item in a new tab. There's a bug here, please fix

jonathanmdinh replied:
added same functionality to instagram handle in commit 427aa9b6b39e2869116a72b7c2b1658270bdf950. please try with this new commit

d-la replied:
I see the issue here - going to resolve this comment and add a new comment @jonathanmdinh

d-la reviewed (changes requested):
Take a look at my comments and address them you when you have a chance, thanks!

jonathanmdinh committed:
added linking out of site functionality to instagram handle

jonathanmdinh committed:
adjusted location of global contact information field retrieval to app.php

d-la commented on wp-content/themes/chrisdinh-photography/resources/acf-json/group_65baaebbb3f1c.json:
```diff
@@ -0,0 +1,93 @@
+            "preview_size": "medium"
+        },
+        {
+            "key": "field_65de2c5c0eefc",
```
This entire field is not necessary (for some reason i'm also not seeing this show up on my local. Might be because our databases aren't synced)

But Wordpress by default supports opening links in a new tab as an option

In the Menus page, click the Screen Options at the top right and click check "Link Target" so that each menu item has the "Open link in a new tab" checkbox. This is the one you should be using since its supported out of the box, no need for an ACF field here @jonathanmdinh 

![Screen Shot 2024-02-29 at 6 04 33 PM](https://github.com/jonathanmdinh/itschrisdinh.com/assets/37962617/7cc1ac1f-b387-4d5a-b526-5e838195bebd)

d-la commented on wp-content/themes/chrisdinh-photography/resources/acf-json/group_65de118a2edd9.json:
```diff
@@ -0,0 +1,163 @@
+            "layout": "block",
+            "sub_fields": [
+                {
+                    "key": "field_65de121ee872a",
```
All of this can be condensed into one field. In ACF there's a field type called "Link". It allows you to add a URL, allows you to toggle opening in a new tab, and allows you to add a Link Title. You should use this field instead of 3 separate fields to control how the link appears. Make sure you update your logic to use the data retrieved from that field to control how the HTML is rendered @jonathanmdinh

jonathanmdinh committed:
converted url fields to links to add target functionality

jonathanmdinh commented:
> Take a look at my comments and address them you when you have a chance, thanks!

In Commit 7160b3b3923d2a490a488f82b2105c816bfe9665 
- Converted URL fields to links to consolidate multiple fields into one. 
- Modified PHP to reference new fields and incorporated "Open link in new tab" functionality that comes with native WordPress into code.

d-la reviewed (approved):
Theres a couple nit picky things i saw but this is looking much better.

Squash your commits and merge this in!
